lib/delocalize/localized_date_time_parser.rb

```diff
@@ -66,7 +66,7 @@ def translate_month_and_day_names(datetime)
         end.flatten.compact
 
         original = (Date::MONTHNAMES + Date::ABBR_MONTHNAMES + Date::DAYNAMES + Date::ABBR_DAYNAMES).compact
-        translated.each_with_index { |name, i| datetime.gsub!(/\b#{name}\b/, original[i]) }
+        translated.each_with_index { |name, i| datetime.dup.gsub!(/\b#{name}\b/, original[i]) }
```

Comment:
Does this actually do anything though?  Aren't you just duping the string, modifying it and then throwing it away when it goes out of scope?

CC @aergonaut

Reply:
Correct. `each_with_index` discards the return value of the block, so calling `dup` here has no effect as the duplicated string is discarded after each iteration. The code worked before because `gsub!` mutates its receiver.

What we could do instead is map over the iterator that `each_with_index` produces:

```suggestion
        translated.each_with_index.map { |name, i| datetime.dup.gsub(/\b#{name}\b/, original[i]) }
```